client/lib/upgrades/actions/domain-management.js

```diff
@@ -48,13 +47,7 @@ function setPrimaryDomain( siteId, domainName, onComplete = noop ) {
 			return onComplete( error, data );
 		}
 
-		sites.setSelectedSite( siteId );
-
-		Dispatcher.handleServerAction( {
-			type: 'FETCH_SITES'
-		} );
-
-		sites.once( 'change', () => {
+		requestSite( siteId )( dispatch ).then( () => {
```

Comment:
We're dispatching here `ActionTypes.PRIMARY_DOMAIN_SET_COMPLETED`, however that might not be true, since [the catch clause](https://github.com/Automattic/wp-calypso/blob/master/client/state/sites/actions.js#L118-L124) might swallowed the error.

Another thing is, entirely minor, that's a bit confusing to me:
Since `requestSite` is an action creator, meaning it's result supposed to be an action ( in this case -> thunk ), so I would've expected this to be called as:
`dispatch( requestSite( siteId ) ).then(...)`, however that might not work if there are any rogue middlewares that don't `return dispatch(...)`